### PR TITLE
Server config

### PR DESCRIPTION
**Pull request by Hexawolf**
I've moved most of "snowflake" things to config so ss-server can already be used outside of Hexamine

### CODE REVIEW

**Comment by foxcpp** [comment]
YAML: https://github.com/go-yaml/yaml. Supports struct tags.
TOML: https://github.com/BurntSushi/toml. Supports struct tags.
INI: https://github.com/go-ini/ini. Supports struct tags.

JSON is bad format for configuration files. 

1. It doesn't supports comments so you can't explain why certain option is to certain value or quickly turn on/off configuration options
2. JSON is not readable as alternatives.

And this is not only my opinion: 
- https://arp242.net/weblog/json_as_configuration_files-_please_dont
- https://dzone.com/articles/why-json-isnt-a-good-configuration-language
- https://dev.mikamai.com/2014/10/06/why-json-sucks-for-user-configuration/
- https://www.lucidchart.com/techblog/2018/07/16/why-json-isnt-a-good-configuration-language/

File: ss-server/config.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			}
+			serverConfig.NewConfig()
+			enc := toml.NewEncoder(configFile)
+			enc.Encode(serverConfig)
+		} else {
+			return err
+		}
+	}
+	defer configFile.Close()
+	_, err = toml.DecodeReader(configFile, c)
```
Comment (foxcpp) [inline ss-server/config.go]: Shouldn't this fail if configFile is opened only for writing at line 91?

Comment (foxcpp) [inline ss-server/config.go]: Shouldn't this fail if configFile is opened only for writing at line 91?

Comment (Hexawolf) [reply ss-server/config.go]: It is opened with O_RDWR, this won't fail unless library developer is retarded.

Comment (foxcpp) [reply ss-server/config.go]: Oh, ok.

**Pull request merged**